### PR TITLE
[tx] Add LoRA to attention layers

### PR DESCRIPTION
**Pull request by tyler-griggs**
Partially addresses #398

* Remove `MultiHeadProj` in favor of using `LoRALinear` for Q/K/V/O projections
* Shard projections along `num_heads * head_dim` axis -- to reconsider as we work on multi-node. vLLM also flattens to `num_heads * head_dim` but also asserts that `num_heads % tp_size == 0`. To start, we will make the same assumption, but to be reconsidered with multi-node.

### CODE REVIEW

File: skyrl-tx/tx/models/qwen3.py [hunk 1]
```diff
@@ -47,41 +35,51 @@ def __init__(self, config: Qwen3Config, *, dtype: jnp.dtype, rngs: nnx.Rngs) ->
...
         self.num_kv_heads = config.num_key_value_heads
         self.head_dim = getattr(config, "head_dim", None) or config.hidden_size // self.num_heads
-
-        self.q_proj = MultiHeadProj(
-            "BMK,KNH->BMNH",
-            config.hidden_size,
-            self.num_heads,
-            self.head_dim,
-            sharding=jax.P(None, "tp", None),
+        self.max_lora_adapters = getattr(config, "max_lora_adapters", 0)
```
Comment (pcmoritz) [inline skyrl-tx/tx/models/qwen3.py]: We don't need these in the forward, right? So no need to assign them to `self`, we can just keep them normal variables -- that's also consistent with the Qwen3MLP layer
